cmd/crank/beta/validate/unknown_fields.go

```diff
@@ -0,0 +1,25 @@
+package validate
+
+import (
+	"fmt"
+	"k8s.io/apiextensions-apiserver/pkg/apiserver/schema"
+	"k8s.io/apiextensions-apiserver/pkg/apiserver/schema/pruning"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"strings"
+)
+
+// validateUnknownFields Validates the resource's unknown fields against the given schema and returns a list of errors.
+func validateUnknownFields(mr map[string]interface{}, sch *schema.Structural) field.ErrorList {
+	opts := schema.UnknownFieldPathOptions{
+		TrackUnknownFieldPaths: true, // to get the list of pruned unknown fields
+	}
+	errs := field.ErrorList{}
+
+	unkFlds := pruning.PruneWithOptions(mr, sch, true, opts)
```

Comment:
style nit: i think we tend to use first letter abbreviations instead of shortening the words in identifiers, so this could just be `uf` instead